### ISSUE TITLE
[feature request] .struct/.estruct

### ISSUE THREAD

**Issue by zxlame** [Dotneteer/spectnetide#111]
I have not found an analogue of structures. If they are not there, you can add them, I used them quite often for in sjasm and it is very convenient.

**Comment by Dotneteer** [comment]
@zxlame, thanks for the suggestion. Implementing structs requires adding totally new mechanisms to the SpectNetIDE assembler, but I guess it is worth the investment. I think it over, and will add a comment to this issue when I have more details or estimates about the implementation :-)

**Comment by zxlame** [comment]
In the sjasm structure, this is actually a named scope with data offsets. Simply explain with an example.
```
      struct exampl
field1 .db 0
field2 .dw 0
field3 .ds 3
    ends
```

Generate labels
exampl =6
exampl.field1 =0
exampl.field2 =1
exampl.field3 =3
And

 `      .ds exampl*32`

Reserved space for 32 object "exampl"
Thanks for your work!

**Comment by Dotneteer** [comment]
@zxlame, the feature is implemented, released in v1.18.0
You can read more details here: https://dotneteer.github.io/spectnetide/documents/structs#article  
If you like my efforts, please star the project in GitHub, or create a review here: https://marketplace.visualstudio.com/items?itemName=Dotneteer.SpectNetIde

**Comment by zxlame** [comment]
Superb
